modules/sample-modules/extend-product-page.md

```diff
@@ -0,0 +1,572 @@
+    ) {
+        parent::__construct($translator, $locales);
+        $this->defaultCurrency = $defaultCurrency;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        parent::buildForm($builder, $options);
+        $builder
+            ->add('custom_price', MoneyType::class, [
+                'label' => $this->trans('My custom price', 'Modules.Demoproductform.Admin'),
+                'label_tag_name' => 'h3',
+                'currency' => $this->defaultCurrency->iso_code,
+                'required' => false,
+                'constraints' => [
+                    new NotBlank(),
+                    new Type(['type' => 'float']),
+                    new PositiveOrZero(),
+                ],
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $resolver
+            ->setDefaults([
+                'label' => $this->trans('Customization', 'Modules.Demoproductform.Admin'),
+            ])
+        ;
+    }
+}
+```
+
+Declare your service in your `config.yml`:
+
+```yml
+services:
+    DemoNewHooks\Form\Type\CustomTabType:
+        class: DemoNewHooks\Form\Type\CustomTabType
+        parent: 'form.type.translatable.aware'
+        public: true
+        arguments:
+            - '@=service("prestashop.adapter.data_provider.currency").getDefaultCurrency()'
+        tags:
+            - { name: form.type }
+```
+
+Add a `use` statement in your `ProductFormModifier`: 
+
+```php
+use DemoNewHooks\Form\Type\CustomTabType;
+```
+
+Finally, add this `CustomTabType` to the `$productFormBuilder` variable in your `modify` method in the `ProductFormModifier`: 
+
+```php
+/**
+ * @param int|null $productId
+ * @param FormBuilderInterface $productFormBuilder
+ */
+public function modify(
+    int $productId,
+    FormBuilderInterface $productFormBuilder
+): void {
+    
+    $idValue = $productId ? $productId->getValue() : null;
+    $customProduct = new CustomProduct($idValue);
+
+    $this->formBuilderModifier->addAfter(
+        $productFormBuilder,
+        'pricing',
+        'custom_tab',
+        CustomTabType::class,
+        [
+            'data' => [
+                'custom_price' => $customProduct->custom_price,
+            ],
+        ]
+    );
+}
+```
+
+{{% notice note %}}
+Any type added directly to the `$productFormBuilder` (and not to a sub tab) will be considered as a tab.
+{{% /notice %}}
+
+## Backwards compatibility for displayAdminProductsExtra hook
+
+A custom Form Type `ExtraModulesType` as been added to {{< minver v="8.1.0" >}}, to allow backwards compatibility for modules implementing `displayAdminProductsExtra` hook.
+
+If a module has registered a hook on `displayAdminProductsExtra`, a custom tab will be added on the new product page, handled by `ExtraModulesType`. 
```

Comment:
```suggestion
If a module registers to `displayAdminProductsExtra` hook, a custom tab will be added on the new product page, handled by `ExtraModulesType`. 
```

Custom tab with all the modules that hook into this hook, right? Not a custom tab only for this one module?